maven/spec/dependabot/maven/version_spec.rb

```diff
@@ -379,6 +376,92 @@
         end
+
+        context "with equivalent separators" do
+          let(:versions) do
+            [
+              { version: "1.0alpha1", other_version: "1.0-a1" },
+              { version: "1.0beta-1", other_version: "1.0-b1" },
+              { version: "1.0milestone1", other_version: "1.0-m1" },
+              { version: "1.0milestone-1", other_version: "1.0-m1" },
+              { version: "1.0rc-1", other_version: "1.0-cr1" },
+              { version: "1.0rc1", other_version: "1.0-cr1" },
+              { version: "1.0ga", other_version: "1.0" },
+              { version: "1-0.ga", other_version: "1.0" },
+              { version: "1.0-final", other_version: "1.0" },
+              { version: "1-0-ga", other_version: "1.0" },
+              { version: "1-0-final", other_version: "1-0" },
+              { version: "1-0", other_version: "1.0" }
+            ]
+          end
+
+          it "returns 0 for all equivalent versions" do
+            versions.each do |v|
+              version = described_class.new(v[:version])
+              other_version = described_class.new(v[:other_version])
+              expect(version <=> other_version).to eq 0
+            end
+          end
+        end
+
+        context "with unequal separators" do
+          let(:version) { described_class.new("1.0alpha.1") }
+          let(:other_version) { described_class.new("1.0-a1") }
+
+          it { is_expected.to eq(1) }
+        end
+
+        context "with long versions" do
+          let(:versions) do
+            [{
+              version: "1.0.0.0.0.0.0",
+              other_version: "1"
+            }, {
+              version: "1.0.0.0.0.0.0x",
+              other_version: "1x"
+            }]
+          end
+
+          it "returns 0 for equivalent versions" do
+            versions.each do |v|
+              version = described_class.new(v[:version])
+              other_version = described_class.new(v[:other_version])
+              expect(version <=> other_version).to eq 0
+            end
+          end
+        end
+
+        context "when ordering versions" do
+          let(:versions) do
+            [
+              described_class.new("NotAVersionSting"),
+              described_class.new("1.0-alpha"),
+              described_class.new("1.0a1-SNAPSHOT"),
+              described_class.new("1.0-alpha1"),
+              described_class.new("1.0beta1-SNAPSHOT"),
+              described_class.new("1.0-b2"),
+              described_class.new("1.0-beta3.SNAPSHOT"),
+              described_class.new("1.0-beta3"),
+              described_class.new("1.0-milestone1-SNAPSHOT"),
+              described_class.new("1.0-m2"),
+              described_class.new("1.0-rc1-SNAPSHOT"),
+              described_class.new("1.0-cr1"),
+              described_class.new("1.0-SNAPSHOT"),
+              described_class.new("1.0"),
+              described_class.new("1.0-sp"),
+              # described_class.new("1.0-a"),
```

Comment:
I need to check with the maven tool if this order with the commented lines is correct

Reply:
Checked and this is where the commented version should be in the sorted list.